x-pack/libbeat/management/fleet/manager.go

```diff
@@ -122,38 +126,52 @@ func (cm *Manager) CheckRawConfig(cfg *common.Config) error {
 	return nil
 }
 
+// UpdateStatus updates the manager with the current status for the beat.
+func (cm *Manager) UpdateStatus(status management.Status, msg string) {
+	cm.lock.Lock()
+	if cm.status != status || cm.msg != msg {
+		cm.status = status
+		cm.msg = msg
+		cm.lock.Unlock()
+		cm.client.Status(statusToProtoStatus(status), msg)
+		cm.logger.Infof("Status change to %s: %s", status, msg)
+		return
+	}
+	cm.lock.Unlock()
```

Comment:
Why unlock the call to `client.Status`? If we have a race between unlock and the Status being send by two go-routines, how can you tell which one is the newer status update?

Reply:
Yeah I was commenting about on that as well, I can move the `client.Status` call inside of the lock. I think that would be the best.